src/controllers/impostersController.js

```diff
@@ -120,6 +120,96 @@ function create (protocols, imposters, Imposter, logger) {
+
+                        var textFinalStored = fs.readFileSync(storeImposterDir + '/store_imposters_' + getStoredFileName[0] + '.json', 'utf-8');
+                        var constructStored = '[' + textFinalStored.slice(0, -1) + ']';
+                        var parseImposterStored = JSON.parse(constructStored);
+                        parseImposterStored.forEach(function (parseStored, index) {
+                            var savePortStored = (parseStored.port).toString();
+                            if (savePortStored === replayablePort) {
+                                parseImposterStored.splice(index, 1);
+                                parseImposterStored.push(saveImposter);
+                            }
+                        });
+                        var eliminateArray = JSON.stringify(parseImposterStored);
+                        var finalArray = eliminateArray.slice(1, -1);
+                        fs.writeFileSync(storeImposterDir + '/store_imposters_' + getStoredFileName[0] + '.json', finalArray + ',');
+                    }
+                });
+            }
+
+        }
+    }
+
+    function saveImposter (imposter) {
+        var fs = require('fs');
+        var mountebank = require('../mountebank');
+        var saveFile = mountebank.saveImpostersFile;
+        var saveFileFlag = mountebank.saveImpostersFileFlag;
+        if ((saveFileFlag) && (saveFile !== undefined)) {
+            var makeString = saveFile.toString();
+            var getStoredFileName = makeString.split('.');
+            var storeImposterDir = './StoreImposters';
+            var ImposterDir = './Repository_Template';
+
+            if (!fs.existsSync(storeImposterDir)) {
+                fs.mkdirSync(storeImposterDir);
+            }
+            if (!fs.existsSync(ImposterDir)) {
+                fs.mkdirSync(ImposterDir);
+            }
+
+            fs.appendFileSync(storeImposterDir + '/store_imposters_' + getStoredFileName[0] + '.json', imposter.trim() + ',');
+            var text = fs.readFileSync(storeImposterDir + '/store_imposters_' + getStoredFileName[0] + '.json', 'utf-8');
+            fs.writeFileSync(ImposterDir + '/' + saveFile, '{"imposters":[' + text.slice(0, -1) + ']}');
+            var textFinal = fs.readFileSync(ImposterDir + '/' + saveFile, 'utf-8');
+            var parseImposter = JSON.parse(textFinal);
+            var myArray = [];
+            var portCollection = [];
+            (parseImposter.imposters).forEach(function (parse) {
+                delete parse._links;
+                var savePort = JSON.stringify(parse.port);
+                if (portCollection.indexOf(savePort) === -1) {
+                    myArray.push(parse);
+                    portCollection.push(savePort);
+                }
+            });
+            fs.writeFileSync(ImposterDir + '/' + saveFile, '{"imposters":' + JSON.stringify(myArray) + '}');
```

Comment:
Since we're paying full I/O and serialization costs even for adding one imposter, we should centralize the logic to do it and just overwrite the entire file with the current state on any state change. Ideas of how to do that below in the mountebank.js comments.

There's a future feature (https://github.com/bbyars/mountebank/issues/177) around improving the test data management, but 1) we can cross that bridge when we get there and 2) it'll be easier to implement with this logic centralized